cmd/flow-dps-live/main.go

```diff
@@ -0,0 +1,342 @@
+	// access node for bootstrapping the peer-to-peer network with other
+	// staked access nodes and unstaked consensus followers. For every finalized
+	// block, it calls the provided callback, which lets the DPS consensus
+	// follower update its data.
+	seedHost, port, err := net.SplitHostPort(flagSeedAddress)
+	if err != nil {
+		log.Error().Err(err).Str("address", flagSeedAddress).Msg("could not parse seed node address")
+		return failure
+	}
+	seedPort, err := strconv.ParseUint(port, 10, 16)
+	if err != nil {
+		log.Error().Err(err).Str("port", port).Msg("could not parse seed node port")
+		return failure
+	}
+	seedKey, err := crypto.DecodePublicKeyHex(crypto.ECDSA_P256, flagSeedKey)
+	if err != nil {
+		log.Error().Err(err).Str("key", flagSeedKey).Msg("could not parse seed node network public key")
+		return failure
+	}
+	seedNode := unstaked.BootstrapNodeInfo{
+		Host:             seedHost,
+		Port:             uint(seedPort),
+		NetworkPublicKey: seedKey,
+	}
+	privKey, err := crypto.DecodePrivateKeyHex(crypto.ECDSA_P256, flagPrivateKey)
+	if err != nil {
+		log.Error().Err(err).Str("key", flagPrivateKey).Msg("could not parse network private key")
+		return failure
+	}
+	follower, err := unstaked.NewConsensusFollower(
+		privKey,
+		flagAddress,
+		[]unstaked.BootstrapNodeInfo{seedNode},
+		unstaked.WithDB(db),
+	)
+	if err != nil {
+		log.Error().Err(err).Str("bucket", flagBucket).Msg("could not create consensus follower")
+		return failure
+	}
+	follower.AddOnBlockFinalizedConsumer(consensus.OnBlockFinalized)
+
+	// Initialize the index writer, which is responsible for writing the chain
+	// and execution data to the index database.
+	write := index.NewWriter(db, storage)
+	defer func() {
+		err := write.Close()
+		if err != nil {
+			log.Error().Err(err).Msg("could not close index writer")
+		}
+	}()
+
+	// Initialize the state transition library, the finite-state machine (FSM)
+	// and then register the desired state transitions with the FSM.
+	transitions := mapper.NewTransitions(log, load, consensus, execution, write,
+		mapper.WithIndexCommit(true),
+		mapper.WithIndexHeader(true),
+		mapper.WithIndexCollections(true),
+		mapper.WithIndexGuarantees(true),
+		mapper.WithIndexTransactions(true),
+		mapper.WithIndexResults(true),
+		mapper.WithIndexEvents(true),
+		mapper.WithIndexPayloads(true),
+		mapper.WithIndexSeals(true),
+	)
+	forest := forest.New()
+	state := mapper.EmptyState(forest)
+	fsm := mapper.NewFSM(state,
+		mapper.WithTransition(mapper.StatusEmpty, transitions.BootstrapState),
+		mapper.WithTransition(mapper.StatusUpdating, transitions.UpdateTree),
+		mapper.WithTransition(mapper.StatusMatched, transitions.CollectRegisters),
+		mapper.WithTransition(mapper.StatusCollected, transitions.IndexRegisters),
+		mapper.WithTransition(mapper.StatusIndexed, transitions.ForwardHeight),
+		mapper.WithTransition(mapper.StatusForwarded, transitions.IndexChain),
```

Comment:
I found it redundant to have two transitions that do exactly the same. Then I realized that if we just return `nil` from any transition, it will just go right back into that transition. Then I realized if we add a short sleep before that return, we don't need an extra transition. So no :D